src/Features/Core/Portable/EditAndContinue/EditAndContinueDocumentAnalysesCache.cs

```diff
@@ -187,21 +200,29 @@ static async (arg, cancellationToken) =>
             {
                 try
                 {
-                    var analyzer = arg.document.Project.Services.GetRequiredService<IEditAndContinueAnalyzer>();
-                    return await analyzer.AnalyzeDocumentAsync(arg.baseProject, arg.self._baseActiveStatements, arg.document, arg.activeStatementSpans, arg.self._capabilities, arg.self._log, cancellationToken).ConfigureAwait(false);
+                    var analyzer = arg.oldProject.Services.GetRequiredService<IEditAndContinueAnalyzer>();
+                    return await analyzer.AnalyzeDocumentAsync(
+                        arg.documentId,
+                        arg.oldProject,
+                        arg.newProject,
+                        arg.self._baseActiveStatements,
+                        arg.activeStatementSpans,
+                        arg.self._capabilities,
+                        arg.self._log,
+                        cancellationToken).ConfigureAwait(false);
                 }
                 catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e, cancellationToken))
                 {
                     throw ExceptionUtilities.Unreachable();
                 }
             },
-            arg: (self: this, document, baseProject, activeStatementSpans));
+            arg: (self: this, documentId, newDocument, oldProject, newProject, activeStatementSpans));
 
         // Previous results for this document id are discarded as they are no longer relevant.
         // The only relevant analysis is for the latest base and document snapshots.
         // Note that the base snapshot may evolve if documents are dicovered that were previously
         // out-of-sync with the compiled outputs and are now up-to-date.
-        _analyses[document.Id] = (lazyResults, baseProject, document, activeStatementSpans);
+        _analyses[documentId] = (lazyResults, oldProject, newDocument, activeStatementSpans);
```

Comment:
I'm looking at the 'before' code and I get that oldProject/newDocument is the same as baseProject/document, but is there a way to enforce it (again, thinking assert, no Release impact).

Reply:
Added some asserts above, but we do not have anything that says "this is old" or "this is new". They are just two different snapshots.